### PR TITLE
[cxx] Strengthen some types away from gpointer.

### PR DESCRIPTION
**Pull request by jaykrell**

### CODE REVIEW

**Comment by jaykrell** [comment]
@monojenkins build Linux AArch64

**Comment by jaykrell** [comment]
@monojenkins build OS X i386

**Comment by jaykrell** [comment]
@monojenkins build Linux AArch64 FullAOT

File: mono/mini/mini-amd64.h [hunk 1]
```diff
@@ -210,16 +210,16 @@ typedef struct MonoCompileArch {
...
+	CallInfo *cinfo;
 	gint32 async_point_count;
-	gpointer vret_addr_loc;
+	MonoInst *vret_addr_loc;
+	MonoInst *seq_point_info_var;
+	MonoInst *ss_tramp_var;
+	MonoInst *bp_tramp_var;
+	MonoInst *lmf_var;
 #ifdef HOST_WIN32
 	struct _UNWIND_INFO* unwindinfo;
```
Comment (jaykrell) [inline mono/mini/mini-amd64.h]: Also moved the "variable" part to the end, so more things are more the same, i.e. the offsets of the common parts are the same on all platforms.

**Comment by jaykrell** [comment]
@monojenkins build  Linux AArch64

**Comment by jaykrell** [comment]
@monojenkins build Linux AArch64

**Comment by jaykrell** [comment]
Review?

**Comment by jaykrell** [comment]
Review?

**Comment by jaykrell** [comment]
We could keep/add casts instead but this is *definitely* the way to go. It cleans up the code & makes it easier to understand & maintain & debug.

**Comment by jaykrell** [comment]
Review?

File: mono/mini/mini-amd64.c [hunk 2]
```diff
@@ -1443,7 +1443,7 @@ mono_arch_compute_omit_fp (MonoCompile *cfg)
 
 	if (!cfg->arch.cinfo)
 		cfg->arch.cinfo = get_call_info (cfg->mempool, sig);
-	cinfo = (CallInfo *)cfg->arch.cinfo;
+	cinfo = cfg->arch.cinfo;
```
Comment (vargaz) [inline mono/mini/mini-amd64.c]: If we forward declare CallInfo, then why are these casts needed ?

Comment (jaykrell) [reply mono/mini/mini-amd64.c]: The intent (did I mess up?) is to remove casts, not add them.
And then to make it more valid C++ in the process -- but better C either way imho.

Comment (vargaz) [reply mono/mini/mini-amd64.c]: Misread it.

Comment (jaykrell) [reply mono/mini/mini-amd64.c]: Oh -- perhaps, you mean -- we could cut out local variables in the process? Yes we probably could.

**Pull request merged**